packages/headless/doc-parser/use-cases/product-listing.ts

```diff
@@ -0,0 +1,56 @@
+import {ActionLoaderConfiguration} from '../src/headless-export-resolvers/action-loader-resolver';
+import {ControllerConfiguration} from '../src/headless-export-resolvers/controller-resolver';
+import {EngineConfiguration} from '../src/headless-export-resolvers/engine-resolver';
+
+const controllers: ControllerConfiguration[] = [
+  {
+    initializer: 'buildProductListing',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildPager',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildResultsPerPage',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildSort',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildFacetManager',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildFacet',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildCategoryFacet',
+    samplePaths: {},
+  },
+  {
+    initializer: 'buildNumericFacet',
+    samplePaths: {},
```

Comment:
It would be good to have these samplePaths filled out, but this can be done in future PRs.

Pointing to existing samples for these controllers is _possible_, but the import paths and certain types would be incorrect. So a developer would not be able to copy-paste the sample (I'm realizing writing this I need to adjust certain paths for the `recommendation` and `product recommendation` use-cases).

In the `headless-react` project, we would need a `ProductListingPage.tsx`. From there we can build it out using a product listing engine, and the correct controller imports.

Reply:
Yeah, I unfortunately found this page yesterday :sweat_smile: Should have searched earlier.

I think there are three scenarios for our samples:

1- Product Listing controller (this one I think it's really important to create a sample)
2- The sort controller, which doesn't front the same properties (so still relevant, but very similar)
3- The rest of the controllers **only** have a different import path (Not sure about those. Technically, only one line changes...)

I'll check with the team if we could create (at the very least) the Product Listing sample somewhere this week